model/scikit/dffml_model_scikit/scikit_base.py

```diff
@@ -119,6 +121,123 @@ def _filename(self):
+                target.append("TRUE_CLUSTER")
+        async for repo in sources.with_features(self.features):
+            feature_data = repo.features(self.features)
+            data.append(feature_data)
+        df = pd.DataFrame(data)
+        xdata = np.array(
+            df.drop(target, axis=1)
+        )  # target is ignored if passed in training data
+        self.logger.info("Number of input repos: {}".format(len(xdata)))
+        self.clf.fit(xdata)
+        joblib.dump(self.clf, self._filename())
+
+    async def accuracy(self, sources: Sources) -> Accuracy:
+        if not os.path.isfile(self._filename()):
+            raise ModelNotTrained("Train model before assessing for accuracy.")
+        data = []
+        target = []
+        estimator_type = getattr(self.clf, "_estimator_type")
+        if estimator_type is "clusterer":
+            if "TRUE_CLUSTER" in set(self.features):
+                target = ["TRUE_CLUSTER"]
+        async for repo in sources.with_features(self.features):
+            feature_data = repo.features(self.features)
+            data.append(feature_data)
+        df = pd.DataFrame(data)
+        xdata = np.array(df.drop(target, axis=1))
+        self.logger.debug("Number of input repos: {}".format(len(xdata)))
+        if "TRUE_CLUSTER" in target:
+            ydata = np.array(df["TRUE_CLUSTER"])
+            if hasattr(self.clf, "predict"):
+                # xdata can be training data or unseen data
+                # inductive clusterer with ground truth
+                y_pred = self.clf.predict(xdata)
+                self.confidence = adjusted_mutual_info_score(ydata, y_pred)
+            else:
+                # requires xdata = training data
+                # transductive clusterer with ground truth
+                self.confidence = adjusted_mutual_info_score(
+                    ydata, self.clf.labels_
+                )
+        else:
+            if hasattr(self.clf, "predict"):
+                # xdata can be training data or unseen data
+                # inductive clusterer without ground truth
+                y_pred = self.clf.predict(xdata)
+                self.confidence = silhouette_score(xdata, y_pred)
+            else:
+                # requires xdata = training data
+                # transductive clusterer without ground truth
+                self.confidence = silhouette_score(xdata, self.clf.labels_)
+        self.logger.debug("Model Accuracy: {}".format(self.confidence))
+        return self.confidence
+
+    async def predict(
+        self, repos: AsyncIterator[Repo]
+    ) -> AsyncIterator[Tuple[Repo, Any, float]]:
+        if not os.path.isfile(self._filename()):
+            raise ModelNotTrained("Train model before prediction.")
+        target = []
+        estimator_type = getattr(self.clf, "_estimator_type")
+        if estimator_type is "clusterer":
+            if "TRUE_CLUSTER" in set(self.features):
+                target.append("TRUE_CLUSTER")
```

Comment:
Let's add another parameter to the config, called `tcluster` which defaults to `None`, if set, use the string given as the feature name which is the `TRUE_CLUSTER`